src/common/SurgeLogger.h

```diff
@@ -0,0 +1,7 @@
+/*
+ * SurgeLogger - quick wrapper to hold reference to and bootstrap the spd logger
+ */
+
+#pragma once
+
```

Comment:
It's a convention used in Linux, and some other open source projects have inherited it: https://www.kernel.org/doc/html/v4.17/process/submitting-patches.html (sections 11, 12 and 13). Not something that I proactively push to this project. Just want to know what kind of conventions we are going to use.

Reply:
Oh gotcha. I’ve never worked on the kernel. (Btw: thanks for working on the kernel!!) But many projects I’ve worked on which have succeeded have well defined rules like that. We should probably adopt them also and make a pull request template.

As you have seen, I’ve squashed all my commits. For this project, that’s a fine thing to do. And the “-i” option to rebase makes it easy enough. In doing so, I think I improved the commit log buy making sure the commits link to github issue and have a clear English explanation. Perfectly reasonable thing to expect.

Do you see any other problems with the PR in the current state? Since it adds a new dep, I’ll still wait for one of @asimilon  @kurasu @kzantow or @whydoubt to take a look I think, just in case they have orthogonal suggestions!

Oh and maybe lets open a fresh issue for “developer documentation” which includes a new doc altogether so we can include things like how to PR, how the branches work, how to set up emacs properly (sorry about those 2 space tabs! Do’h) and so on.  Like I’ve said several times, I’m happy to work any way we want. I think we probably need less heavyweight tools than the Linux kernel but a standard would be lovely.

Reply:
Sounds like a plan :)

Reply:
Great!